services/bots.py

```diff
@@ -13,15 +14,46 @@
 log = logging.getLogger(__name__)
 
 
+def get_owner_installation_id(
+    owner: Owner,
+    deprecated_using_integration: bool,
+    *,
+    repository: Optional[Repository] = None,
+    installation_name: str = GITHUB_APP_INSTALLATION_DEFAULT_NAME,
+) -> Optional[int]:
+    default_app_installation_filter = filter(
+        lambda obj: obj.name == installation_name, owner.github_app_installations or []
+    )
+    # filter is an Iterator, so we need to scan matches
+    # in practice we only consider the 1st one
+    for app_installation in default_app_installation_filter:
```

Comment:
will this always contain one or zero installations? or is there a possibility of there being multiple?

Reply:
Currently there will always be 0 or 1 installations I believe. That comes from Github when the user installs the codecov gh app to their profile. I don't think it's possible for a user in GH to have multiple installations of the same app in their profile.

Assume in the future there are many such installations for some reason.
The idea of `installation_name` is to identify app's functionalities in such a way that they are interchangeable.
Although we'd probably want to be able to pass a list of names and select one option randomly... but that's for the future and there might be other implementation choices available.